src/sqlfluff/rules/L016.py

```diff
@@ -443,7 +443,15 @@ def _compute_source_length(cls, segments: Sequence[BaseSegment]) -> int:
             # source slice, and if we didn't correct for this, we'd count the
             # length of {{bi_ecommerce_orders}} roughly 10 times, resulting in
             # vast overcount of the source length.
-            if slice not in seen_slices:
+            #
+            # :TRICKY: New segments (i.e. those introduced by earlier fixes)
+            # have empty source slices. We definitely want to count the length
+            # of these segments. We can be sure they aren't the tricky templated
+            # segment case described above because new segments are never templated
+            # (because "sqlfluff fix" produced them, not the templater!).
+            if (
+                slice[0] == slice[1] and not segment.is_meta
+            ) or slice not in seen_slices:
```

Comment:
This is much nicer! And easier to understand.

Do wonder if there's any other zero-length segments like meta? Are comments?

Reply:
Not sure. Note that if they're 0-length, it's safe to process them anyway. I added `is_meta` more to clarify the intent. It works fine either way.

Reply:
I checked a comment in the debugger. Length works as expected (not 0).
```
(Pdb) this_line[-1]
<CommentSegment: ([L:  6, P:  1]) "-- The following is the slope of the regression line. Note that CORR (which is the Pearson's correlation">
(Pdb) this_line[-1].raw
"-- The following is the slope of the regression line. Note that CORR (which is the Pearson's correlation"
(Pdb) self._compute_segment_length(this_line[-1])
104

```